Tests/SwiftKafkaTests/KafkaProducerTests.swift

```diff
@@ -110,7 +110,7 @@ final class KafkaProducerTests: XCTestCase {
             }
 
             let expectedTopic = "test-topic"
-            let message = KafkaProducerMessage(
+            let message = KafkaProducerMessage<ByteBuffer, ByteBuffer>(
```

Comment:
Keys are optional, which leads to the problem that their type cannot be inferred when set to `nil`. This is a workaround but far from perfect. Any suggestions?

Reply:
Can we conform `Never` to `KafkaBuffer` and provide an init that doesn't take a key when we are `Never`.